### PR TITLE
Feat: run lock

### PR DESCRIPTION
**Pull request by karitra**
Try to take cluster-wide lock from not too large set before attempt to start any application, in case of failure of lock acquisition attempt on next Orca round.  

**TODO:**
 - [x] update tests to repair coverage and after commit tests.
 - [x] version bump.

### CODE REVIEW

File: src/cocaine/burlak/semaphore.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        self._sharding = sharding
+        self._unicorn = unicorn
+        self._locks = {}
+
+    @gen.coroutine
+    def make_lock_path(self, i):
+        _uuid, path = yield self._sharding.get_semaphore_route()
+        lock_name = self._config.semaphore.lock_name
+
+        raise gen.Return('{}/{}{}'.format(path, lock_name, i))
```
Comment (shaitan) [inline src/cocaine/burlak/semaphore.py]: Lets move `.` from `lock_name` to format string `'{}/{}.{}'`

File: src/cocaine/burlak/burlak.py [hunk 2]
```diff
@@ -837,6 +839,8 @@ def process_loop(self):
         control_filter = yield self.get_filter_once()
         no_state_yet = True
 
+        run_lock = LockHolder()
```
Comment (shaitan) [inline src/cocaine/burlak/burlak.py]: Could `LockHolder` wrapper be dropped? And use underlying lock as is:
```python
run_lock = None
# ...
run_lock = yield semaphore.release_lock_holder(run_lock)
# ...
if run_lock is None:
    # ...
    run_lock = yield semaphore.try_to_acquire_lock()
# ...
```

Comment (karitra) [reply src/cocaine/burlak/burlak.py]: No, it is needed as keeper of object, which would be reseted to None from another class to signal `start_app` completion.

File: src/cocaine/burlak/semaphore.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                lock_holder.lock = Semaphore.Lock(ch)
+
+                raise gen.Return(lock_holder.lock)
+
+        except gen.TimeoutError:
+            self.debug('timeout on lock, probably it is taken already')
+
+        self.info('failed to get lock {}, {}', lock_to_try, lock_path)
+
+        yield seld._close_safe(ch)
```
Comment (kletkavrubashku) [inline src/cocaine/burlak/semaphore.py]: self?

Comment (karitra) [reply src/cocaine/burlak/semaphore.py]: Fixed.

**Pull request merged**